### PR TITLE
Update host health state based on saptune data

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Use saptune state to change host health.
- If SAP workload is found in the host (meaning, a database or application instance is there):
    - saptune not installed -> warning
    - unsupported version -> warning
    - not compliant -> critical
    - not tuned -> warning
    - compliant -> passing
- If SAP workload is not found in the host:
    - passing 

In order to achieve this, I have added a new `is_sap_running` field to the `UpdateSaptuneStatus`. This field is used to determine if a SAP workload is there.
Together with this, we need store the `saptune_health` in the aggregate and create a new event to update it. Very similar of what we do with the checks health.

## How was this tested?

UT

### CODE REVIEW

**Review by EMaksy — APPROVED** [review]
Hey man great job, codewise lgtm :D

**Review by CDimonaco — APPROVED** [review]
LGTM

File: lib/trento/application/integration/discovery/discovery.ex [hunk 1]
```diff
@@ -137,7 +137,13 @@ defmodule Trento.Integration.Discovery do
     ClusterPolicy.handle(event, current_cluster_id)
   end
 
-  defp do_handle(%{"discovery_type" => "saptune_discovery"} = event), do: HostPolicy.handle(event)
+  defp do_handle(%{"discovery_type" => "saptune_discovery", "agent_id" => agent_id} = event) do
```
Comment (rtorrero) [inline lib/trento/application/integration/discovery/discovery.ex:140]: super nitpicky comment but I guess we could save a concatenation by checking the sum of the length of each line?

File: test/trento/domain/host/host_test.exs [hunk 2]
```diff
@@ -944,6 +965,243 @@ defmodule Trento.HostTest do
         end
       )
     end
+
+    test "should update saptune health to passing when a sap workload is removed and saptune is not installed" do
```
Comment (rtorrero) [inline test/trento/domain/host/host_test.exs:969]: sap -> SAP :eyes:

**Review by rtorrero — APPROVED** [review]
LGTM, two minor comments that are just optional changes if you agree :+1:

**Pull request merged**